### ISSUE TITLE
[Modern] relay-compiler has no option to define file-extensions (.jsx)

### ISSUE THREAD

**Issue by olve** [facebook/relay#1671]
.jsx is a widely-used extension for react-components, even when exporting a relay fragment-container from the same file

##### SomeComponent.jsx:
```javascript
class SomeComponent extends React.Component { ... }

export default createFragmentContainer(
    SomeComponent,
    fragments
)
```

relay-compiler enforces .js file-extensions for any relay fragment-containers with no option to allow the common .jsx extension, and the example file above would not compile unless it's file-extension was .js

**Comment by ekosz** [comment]
There is an open PR for this here: https://github.com/facebook/relay/pull/1658

**Comment by kassens** [comment]
fixed by #1658
